File: bone-blueprint/src/main/java/com/bone/blueprint/domain/gateway/InventoryGateway.java
package com.bone.blueprint.domain.gateway;

public interface InventoryGateway {

  /** 校验库存是否充足（创建订单前置条件）。 */
  boolean checkStock(Long productId, Integer quantity);

  /** 创建订单后预留库存（最终一致：支付确认 / 取消释放）。 */
  void reserveStock(Long orderId, Long productId, Integer quantity);

  /** 支付成功后确认扣减（消费预留）。 */
  void confirmStock(Long orderId, Long productId, Integer quantity);

  /** 取消订单后释放预留。 */
  void releaseStock(Long orderId);
}

File: bone-engine/go-engine/bone-blueprint-go/domain/gateway/inventory_gateway.go
package gateway

import (
	"context"
)

// InventoryGateway 库存网关接口
type InventoryGateway interface {
	// CheckStock 检查库存
	CheckStock(ctx context.Context, productID int64, quantity int) (bool, error)

	// ReserveStock 预留库存
	ReserveStock(ctx context.Context, productID int64, quantity int) error

	// ReleaseStock 释放库存
	ReleaseStock(ctx context.Context, productID int64, quantity int) error

	// DeductStock 扣减库存
	DeductStock(ctx context.Context, productID int64, quantity int) error
}
